### PR TITLE
kubeconfig defaults for virtctl

### LINKED ISSUE #635: virtctl should pickup kubectl config
If I launch virtctl, then it does not pickup my kubectl config by default.

This issue is about letting virtctl directly use kubectl's config if available.

Quote:
```
<rmohr> fabiand: regarding to picking up the kubectl context, I think we just need to check for the KUBECONFIG environment variable and fall back to ~/.kube/config in virtctl
```

### PR DESCRIPTION
**Pull request by davidvossel**
fixes #635

### CODE REVIEW

File: pkg/virtctl/virtctl.go [hunk 1]
```diff
@@ -35,8 +35,18 @@ type App interface {
...
 }
 
-func (o *Options) FlagSet() *flag.FlagSet {
+func DefaultKubeconfig() string {
+	kubeConfig := os.Getenv("kubeconfig")
+	if kubeConfig == "" {
+		kubeConfig = os.Getenv("KUBECONFIG")
+	}
+	if kubeConfig == "" {
+		kubeConfig = "~/.kube/config"
```
Comment (stu-gott) [inline pkg/virtctl/virtctl.go]: What would you think about using the `os/user` module to actually determine the correct home dir to use instead of "~"?

Comment (davidvossel) [reply pkg/virtctl/virtctl.go]: yup, good idea. I didn't realize the home dir was something the os.User object had in golang, but it definitely does now that I'm looking at it.

**Comment by davidvossel** [comment]
@rmohr 

I'm not sure how I feel about defaulting the config to $HOME/.kube/config.  I think leaving both the kubeconfig and server options empty is a valid use case for virtctl.

When that occurs, the k8s api package automatically looks for a service account and tries to dynamically make a config.

**Comment by rmohr** [comment]
> I'm not sure how I feel about defaulting the config to $HOME/.kube/config. I think leaving both the kubeconfig and server options empty is a valid use case for virtctl.

Yes this is indeed something to consider. Here my observations:

If I don't have KUBECONFIG specified and no -kubeconfig and no -master, the following happens:

```bash
kubectl get pods -v 9
I0109 09:26:03.161860   25109 round_trippers.go:417] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: .kubectl/v1.9.1 (linux/amd64) kubernetes/3a1c944" http://localhost:8080/api
I0109 09:26:03.162478   25109 round_trippers.go:436] GET http://localhost:8080/api  in 0 milliseconds
```

It seems to fallback to a `http://localhost:8080/api`. By the way I think that there also exists something like a KUBECONFIG_MASTER environment varibale. Further KUBECONFIG can contain a list of configuration files which are getting merged by `kubectl`, as far as I know.

If `~/.kube/config` exists I can see this behaviour:

```bash
cluster/vagrant/.kubectl -v 9 get pods
I0109 09:27:16.613956   25211 loader.go:357] Config loaded from file /home/rmohr/.kube/config
I0109 09:27:16.614142   25211 round_trippers.go:417] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: .kubectl/v1.9.1 (linux/amd64) kubernetes/3a1c944" http://localhost:8080/api
```
It tries to load ~/.kube/config. As a sidenote, in this case the config was just an empty file, so it falls back again to `http://localhost:8080/api`.

> When that occurs, the k8s api package automatically looks for a service account and tries to dynamically make a config.

Yes `kubectl` supports that too, but it is  the last option it checks.

**Comment by davidvossel** [comment]
closed in favor of #648

**Pull request closed without merging**